pkg/endpoint/events.go

```diff
@@ -119,6 +120,71 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	e := ev.ep
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	// If this endpoint is going away, remove all related NOTRACK rules.
+	if err := e.lockAlive(); err != nil {
+		if e.NoTrackPort != "" {
+			if e.IPv4.IsSet() {
+				iptables.RemoveNoTrackRules(e.IPv4.String(), e.NoTrackPort, false)
+			}
+			if e.IPv6.IsSet() {
+				iptables.RemoveNoTrackRules(e.IPv6.String(), e.NoTrackPort, true)
```

Comment:
These functions are not checking the returned errors, but the errors should not be ignored (also below, other callers of these functions).

Reply:
Particular weird corner case here would be if for example IPv4 succeeds but IPv6 fails. Should we then roll back the IPv4 iptables rule configuration since the annotation couldn't be successfully applied? Maybe we decide no, it's fine, if both IPv4 and IPv6 are enabled in the configuration and IPv6 iptables insertion fails then we will fail over to configuring IPv4 connectivity. At least there should be a log message to warn about such a failure so that the functionality doesn't just silently fail in the background with no pointer as to what went wrong.

Reply:
Added log in the underlying `InstallNoTrackRules` and `RemoveNoTrackRules` function. For nodelocaldns, these failures are tolerable, it means DNS traffic WILL be tracked but that's fine in terms of functionality. Maybe the error code can be used later for other cases where this is more failing to install such rules is more critical.

Reply:
That seems reasonable, but in addition I suggest removing the error return code, or using `_ = iptables.Remove...` here.  Stylistically, it's an antipattern to have a function return an error, then have the caller always ignore the error. Down the road, when a developer reads such code, it is impossible to determine whether the error was unintentionally neglected or whether it was intentionally ignored.

Reply:
Thanks! Added logic to process the returned error code and log a warning.